### DISCUSSION TITLE
Can't open zip file after rezipping it with fflate

### DISCUSSION THREAD

**Discussion by realmayus** [101arrowz/fflate#32] [Q&A] [answered]
Hi, I'm trying to unzip an uploaded file and rezip it afterwards. I'm using an input (type=file) and this code gets called when the file changes:
```jsx
<input className={styles.input} type="file" onChange={e => setFile(e.target.files[0])}/>
...

import * as fflate from "fflate";

...
    if (file != null) {
      const fr = new FileReader(file);
      fr.addEventListener('load', (e) => {
        const uint8 = new Uint8Array(e.target.result);
        let unzipped = fflate.unzipSync(uint8);
        const newZip = fflate.zipSync(unzipped);
        const blob = new Blob([newZip], {type: "application/zip"});
        const objectUrl = URL.createObjectURL(blob);
        window.open(objectUrl);
      })
      fr.readAsArrayBuffer(file);
```
Somehow, I can't open the file anymore after downloading it. I also noticed that the output file is significantly bigger than the original one. What could be the reason?

![Screenshot 2021-01-12 at 13 19 30](https://user-images.githubusercontent.com/38134006/104313764-cf266080-54d8-11eb-9b7c-4b233ceaa8f3.png)
![Screenshot 2021-01-12 at 13 19 56](https://user-images.githubusercontent.com/38134006/104313812-dea5a980-54d8-11eb-8c79-1300b49118b7.png)

**Comment by 101arrowz** [comment, marked as answer]
My mistake, I was forgetting to add the data into the file in the synchronous ZIP function, but I only tested the asynchronous version. Will push an update soon.

**Reply by 101arrowz** [reply]
I've updated the package with v0.5.1 on NPM, which fixes this bug. Let me know if you have any issues, and thanks for reporting the problem.
